react/features/display-name/components/DisplayName.js

```diff
@@ -0,0 +1,237 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import { translate } from '../../base/i18n';
+import { participantDisplayNameChanged } from '../../base/participants';
+
+/**
+ * React {@code Component} for displaying and editing a participant's name.
+ *
+ * @extends Component
+ */
+class DisplayName extends Component {
+    /**
+     * {@code DisplayName}'s property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * Whether or not the display name should be editable on click.
+         */
+        allowEditing: React.PropTypes.bool,
+
+        /**
+         * Invoked to update the participant's display name.
+         */
+        dispatch: React.PropTypes.func,
+
+        /**
+         * The participant's current display name.
+         */
+        displayName: React.PropTypes.string,
+
+        /**
+         * A string to append to the displayName, if provided.
+         */
+        displayNameSuffix: React.PropTypes.string,
+
+        /**
+         * The ID attribute to add to the component. Useful for global querying
+         * for the component by legacy components and torture tests.
+         */
+        idAttribute: React.PropTypes.string,
+
+        /**
+         * The ID of the participant whose name is being displayed.
+         */
+        participantID: React.PropTypes.string,
+
+        /**
+         * Invoked to obtain translated strings.
+         */
+        t: React.PropTypes.func
+    };
+
+    /**
+     * Initializes a new {@code DisplayName} instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            /**
+             * Whether or not the component should be displaying an editable
+             * input.
+             *
+             * type {boolean}
+             */
+            isEditing: false,
+
+            /**
+             * The display name to show in the input field while editing the
+             * known display name.
+             */
+            pendingDisplayName: ''
```

Comment:
It's the current entered value in the edit field. It gets set to the participant's display name when editing is begun or is set as an empty string. I can see why the empty string then could be confusing.